debug_toolbar/panels/history/panel.py

```diff
@@ -0,0 +1,75 @@
+import json
+import logging
+from collections import OrderedDict
+
+from django.conf.urls import url
+from django.template.loader import render_to_string
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+
+from debug_toolbar import settings as dt_settings
+from debug_toolbar.panels import Panel
+from debug_toolbar.panels.history import views
+from debug_toolbar.panels.history.forms import HistoryStoreForm
+
+logger = logging.getLogger(__name__)
+
+CLEANSED_SUBSTITUTE = "********************"
+
+
+class HistoryPanel(Panel):
+    """ A panel to display History """
+
+    title = _("History")
+    nav_title = _("History")
+    template = "debug_toolbar/panels/history.html"
+
+    @property
+    def is_historical(self):
+        """The HistoryPanel should not be included in the historical panels."""
+        return False
+
+    @classmethod
+    def get_urls(cls):
+        return [
+            url(r"^history_sidebar/$", views.history_sidebar, name="history_sidebar"),
+        ]
+
+    @property
+    def nav_subtitle(self):
+        return self.get_stats().get("request_url", "")
+
+    def generate_stats(self, request, response):
+        cleansed = request.POST.copy()
+        for k in cleansed:
+            cleansed[k] = CLEANSED_SUBSTITUTE
```

Comment:
Django contains a cleaning machinery somewhere which only substitutes secrets and keys. Might it be worth it reusing that here?

Reply:
There is. Think we should use the [private method](https://github.com/django/django/blob/09341856ed9008875c1cc883dc0c287670131458/django/contrib/auth/__init__.py#L40) or copy it?

Reply:
I copied the method out of Django. You were absolutely right, sanitizing all values was unbearable to look at.